ext/drmgr/drmgr.c

```diff
@@ -567,23 +590,51 @@ cblist_shift_and_resize(cb_list_t *l, uint insert_at)
     return insert_at;
 }
 
+static void
+cblist_insert_other(cb_list_t *l, cb_list_t *l_to_copy)
+{
+    ASSERT(l->entry_sz == l_to_copy->entry_sz, "must be of the same size");
+    size_t i;
+    for (i = 0; i < l_to_copy->num_def; i++) {
+        cb_entry_t *e = &l_to_copy->cbs.bb[i];
+        if (!e->pri.valid)
+            continue;
+        int idx = priority_event_add(l, &e->pri.in_priority);
+        if (idx >= 0) {
+            cb_entry_t *new_e = &l->cbs.bb[idx];
+            *new_e = *e;
+        }
+    }
+}
+
+/* A helper function to help with the copying of cb list data for
+ * cblist_create_local() and cblist_create_global().
+ */
+static void
+cblist_copy(cb_list_t *src, cb_list_t *dst)
+{
+    ASSERT(src->num_def <= dst->capacity, "dst must have large enough capacity");
+    dst->entry_sz = src->entry_sz;
+    dst->num_def = src->num_def;
+    memcpy(dst->cbs.array, src->cbs.array, src->num_def * src->entry_sz);
+}
+
 /* Creates a temporary local copy, using local if it's big enough but
- * otherwise allocating new space on the heap.
- * Caller must hold read lock.
+ * otherwise allocating new space on the heap. It does a scoped allocation
```

Comment:
"scoped allocation" sounds to me like sthg local to this function which is inaccessible on return: do you mean a thread-local heap allocation?

Reply:
Done.